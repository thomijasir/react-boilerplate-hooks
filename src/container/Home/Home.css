@import '../../assets/style/animate.css';

h1 {
  font-family: 'Courier New', Courier, monospace;
}
.rotate {
  animation: rotation 3s infinite linear;
}
@keyframes t {
  from {
    background-size: 0 200%;
  }
}

@keyframes b {
  50% {
    background-position: 0 -100%, 0 0;
  }
}
.typing-master {
  font-family: monospace;
  font-size: 16px;
  color: #0000;
  background: linear-gradient(-90deg, #00dffc 3px, #0000 0) 6px 0,
    linear-gradient(#00dffc 0 0) 0 0;
  background-size: calc(90 * 1ch) 200%;
  -webkit-background-clip: padding-box, text;
  background-clip: padding-box, text;
  background-repeat: no-repeat;
  animation: b 0.7s infinite steps(1), t calc(90 * 0.2s) steps(90) forwards;
}

.credits {
  margin-top: 24px;
  font-size: 12px;
}

