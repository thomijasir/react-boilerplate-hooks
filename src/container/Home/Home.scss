@import '../../assets/style/animate';
.home-page {
  h1 {
    font-family: 'Courier New', Courier, monospace;
  }
  line-height: 1.5;
  height: 100vh;
  display: flex;
  align-items: center;
  text-align: center;
  .rotate {
    animation: rotation 3s infinite linear;
  }
}
