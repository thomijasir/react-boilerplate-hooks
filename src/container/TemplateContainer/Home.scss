.home-page {
  h1 {
    font-family: 'Courier New', Courier, monospace;
  }
  line-height: 1.5;
  height: 100vh;
  display: flex;
  align-items: center;
  text-align: center;
}
