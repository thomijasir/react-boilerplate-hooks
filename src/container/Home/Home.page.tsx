import React, { FC, useState, useEffect, useContext } from 'react';
import Image from '../../components/Image/Image.comp';
import { AppContext } from '../../store/AppProvider';
import { TITLE_HOME_PAGE } from '../../constants';
import reactLogo from '../../assets/images/react.svg';
import './Home.css';

export interface IProps {}

const Home: FC<IProps> = () => {
  const context = useContext(AppContext);
  const [title] = useState('Boilerplate');

  useEffect(() => {
    // eslint-disable-next-line no-console
    console.log(TITLE_HOME_PAGE);
    // eslint-disable-next-line no-console
    console.log('HOME CONTEXT: ', context);
  }, []);

  return (
    <div className="home-page w-full h-full mt-40">
      <div className="w-full text-center">
        <Image
          src={reactLogo}
          className="rotate w-60 h-60 m-auto"
          alt="react-logo"
        />
        <h1 className="text-2xl font-bold p-4">{title}</h1>
        <span className="typing-master text-base xs:text-xs sm:text-base">
          Project Template with the latest react hook, <br /> simple clean and
          no vulnerability package.
        </span>
        <p className="credits text-sm">
          React with tailwind flavour by <b>Thomi Jasir</b>
        </p>
      </div>
    </div>
  );
};

export default Home;
