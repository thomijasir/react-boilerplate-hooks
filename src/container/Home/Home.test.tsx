import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from './Home.page';

afterEach(() => {
  jest.clearAllMocks();
});

describe('HomePage Container Snap Test', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Home />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
