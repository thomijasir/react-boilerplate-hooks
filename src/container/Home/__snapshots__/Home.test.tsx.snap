// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Container Snap Test render() renders the component 1`] = `
<div
  className="home-page safe-area"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <img
          alt="react-logo"
          className="rotate"
          src="react.svg"
        />
        <h1>
          React Boilerplate
        </h1>
        <p>
          Project Template with the latest react hook, simple clean and no vulnerability package
        </p>
        <small>
          React Template by Thomi Jasir
        </small>
      </div>
    </div>
  </div>
</div>
`;
